packages/rs-drive-abci/src/rpc/core.rs

```diff
@@ -1,27 +1,110 @@
 
+/// Information returned by QuorumListExtended
+pub type QuorumListExtendedInfo = HashMap<QuorumHash, ExtendedQuorumDetails>;
+
+/// Core height must be of type u32 (Platform heights are u64)
+pub type CoreHeight = u32;
 /// Core RPC interface
-#[cfg_attr(feature = "fixtures-and-mocks", automock)]
+#[automock]
 pub trait CoreRPCLike {
     /// Get block hash by height
-    fn get_block_hash(&self, height: u32) -> Result<BlockHash, Error>;
+    fn get_block_hash(&self, height: CoreHeight) -> Result<BlockHash, Error>;
+
+    /// Get block hash by height
+    fn get_best_chain_lock(&self) -> Result<CoreChainLock, Error>;
+
+    /// Get transaction
+    fn get_transaction(&self, tx_id: &Txid) -> Result<Transaction, Error>;
+
+    /// Get transaction
+    fn get_transaction_extended_info(&self, tx_id: &Txid) -> Result<GetTransactionResult, Error>;
+
+    /// Get block by hash
+    fn get_fork_info(&self, name: &str) -> Result<Option<Bip9SoftforkInfo>, Error>;
 
     /// Get block by hash
     fn get_block(&self, block_hash: &BlockHash) -> Result<Block, Error>;
 
     /// Get block by hash in JSON format
     fn get_block_json(&self, block_hash: &BlockHash) -> Result<Value, Error>;
+
+    /// Get chain tips
+    fn get_chain_tips(&self) -> Result<GetChainTipsResult, Error>;
+
+    /// Get list of quorums at a given height.
+    ///
+    /// See <https://dashcore.readme.io/v19.0.0/docs/core-api-ref-remote-procedure-calls-evo#quorum-listextended>
+    fn get_quorum_listextended(
+        &self,
+        height: Option<CoreHeight>,
+    ) -> Result<ExtendedQuorumListResult, Error>;
+
+    /// Get quorum information.
+    ///
+    /// See <https://dashcore.readme.io/v19.0.0/docs/core-api-ref-remote-procedure-calls-evo#quorum-info>
+    fn get_quorum_info(
+        &self,
+        quorum_type: QuorumType,
+        hash: &QuorumHash,
+        include_secret_key_share: Option<bool>,
+    ) -> Result<QuorumInfoResult, Error>;
+
+    /// Get the difference in masternode list, return masternodes as diff elements
+    fn get_protx_diff_with_masternodes(
+        &self,
+        base_block: Option<u32>,
+        block: u32,
+    ) -> Result<MasternodeListDiff, Error>;
+
+    // /// Get the detailed information about a deterministic masternode
+    // fn get_protx_info(&self, pro_tx_hash: &ProTxHash) -> Result<ProTxInfo, Error>;
 }
 
 /// Default implementation of Dash Core RPC using DashCoreRPC client
 pub struct DefaultCoreRPC {
     inner: Client,
 }
 
+macro_rules! retry {
+    ($action:expr) => {{
+        const MAX_RETRIES: u32 = 4; // Maximum number of retry attempts
+        const FIB_MULTIPLIER: u64 = 1; // Multiplier for Fibonacci sequence
+
+        fn fibonacci(n: u32) -> u64 {
+            match n {
+                0 => 0,
+                1 => 1,
+                _ => fibonacci(n - 1) + fibonacci(n - 2),
+            }
+        }
+
+        let mut last_err = None;
+        for i in 0..MAX_RETRIES {
```

Comment:
We need to handle only connection error here, otherwise even invalid input will lead to retries